### PR TITLE
Fix Component Tests

### PR DESCRIPTION
**Pull request by ryanrath**
## Description
- Fixing the component tests (Acl related) so that they utilize the new
  environment specific location for files that contain the expected results.

## Motivation and Context
Not broken tests are good.

## Tests performed
the tests?

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php [hunk 1]
```diff
@@ -1297,4 +1355,78 @@ public function provideGetFormalRoleName()
...
+        while (array_key_exists($username, self::$users)) {
+            $suffix = rand(self::MIN_USERS, self::MAX_USERS);
+            $username = "$username$suffix";
+        }
+        return $username;
+    }
+
+    public static function getTestFile($fileName)
+    {
+        if (!isset(self::$ENV)){
```
Comment (plessbd) [inline open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php]: is this actually needed?  for any of this to get called doesnt setUpBeforeClass get called which should initialize this variable?

Comment (ryanrath) [reply open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php]: Yup, if this function is to be called outside of this class then we need to ensure that the environment variable is set before returning a value.

**Comment by plessbd** [comment]
seems like this is failing...
```
There was 1 failure:

1) ComponentTests\XDUserTest::testCreateUserWithNonStandardPrimaryRole
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'mgr'
+'usr'

/root/src/github.com/ubccr/xdmod/open_xdmod/modules/xdmod/component_tests/lib/XDUserTest.php:629

FAILURES!
Tests: 170, Assertions: 289, Failures: 1.
```

**Comment by ryanrath** [comment]
Yup, it was but is passing now.

**Pull request merged**